magento/magento2
# Added validation for admin system configuration

**Add validation for admin system configuration fields in GraphQl and WebAPI system.xml**

### Description (*)

Added validation for admin system configuration fields in `app/code/Magento/GraphQl/etc/adminhtml/system.xml` and `app/code/Magento/Webapi/etc/adminhtml/system.xml`:

1. **GraphQl system.xml**: Added `<validate>validate-zero-or-greater validate-digits</validate>` to the `maximum_page_size` field to ensure only non-negative whole numbers are accepted for maximum page size in paginated GraphQL search results.

2. **WebAPI system.xml**: Added `<validate>validate-zero-or-greater validate-digits</validate>` to three fields:
   - `complex_array_limit` (Input List Limit) - Maximum number of items allowed in an entity's array property
   - `maximum_page_size` (Maximum Page Size) - Maximum number of items allowed in a paginated search result
   - `default_page_size` (Default Page Size) - Default number of items in a paginated search result

These validations prevent invalid input (negative numbers, decimals, non-numeric characters) that could cause runtime errors or API security issues.

### Related Pull Requests
None

### Fixed Issues (if relevant)

1. Fixes magento/magento2# N/A

### Manual testing scenarios (*)

1. Navigate to **Stores → Configuration → Catalog → GraphQl** section in Admin
2. Locate the **Maximum Page Size** field
3. Enter invalid values and verify validation error appears:
   - Negative number (e.g., `-5`)
   - Decimal number (e.g., `10.5`)
   - Non-numeric characters (e.g., `abc`)
4. Enter valid values and verify they save successfully:
   - Zero (`0`)
   - Positive integer (e.g., `100`)
5. Navigate to **Stores → Configuration → Services → Magento Web API** section in Admin
6. Ensure **Input Limiting** is enabled (`input_limit_enabled=1`)
7. Locate **Input List Limit**, **Maximum Page Size**, and **Default Page Size** fields
8. Repeat steps 3-4 for each field to verify validation works correctly

### Questions or comments
The `complex_array_limit` and `maximum_page_size` fields in WebAPI system.xml are within a `<depends>` block that checks `input_limit_enabled=1`, so validation only applies when input limiting is enabled. The `default_page_size` field is not dependent on input limiting.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#40647: Added validation for admin system configuration
